### PR TITLE
feat: Add ed-25519 key provider

### PR DESCRIPTION
**Pull request by amonsosanz**
This PR adds an `ed-25519` key provider.

### CODE REVIEW

File: src/kms/key-providers/ed25519-provider.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  async newPrivateKeyFromSeed(seed: Uint8Array): Promise<KmsKeyId> {
+    if (seed.length !== 32) {
+      throw new Error('Seed should be 32 bytes');
+    }
+
+    const privateKey = seed;
+    const publicKey = ed25519.getPublicKey(privateKey);
+    const kmsId = {
+      type: this.keyType,
+      id: providerHelpers.keyPath(this.keyType, Buffer.from(publicKey).toString('hex'))
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/kms/key-providers/ed25519-provider.ts]: Done

File: src/kms/key-providers/ed25519-provider.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  /**
+   * Gets public key by kmsKeyId
+   * @param {KmsKeyId} keyId - key identifier
+   * @returns {Promise<string>} Public key as a hex string
+   */
+  async publicKey(keyId: KmsKeyId): Promise<string> {
+    const privateKeyHex = await this.privateKey(keyId);
+    const privateKey = Buffer.from(privateKeyHex, 'hex');
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/kms/key-providers/ed25519-provider.ts]: Done

File: src/kms/key-providers/ed25519-provider.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * signs prepared payload of size,
+   * with a key id
+   * @param {KmsKeyId} keyId  - key identifier
+   * @param {Uint8Array} data - data to sign (32 bytes)
+   * @returns {Promise<Uint8Array>} signature
+   */
+  async sign(keyId: KmsKeyId, data: Uint8Array): Promise<Uint8Array> {
+    const privateKeyHex = await this.privateKey(keyId);
+    const privateKey = Buffer.from(privateKeyHex, 'hex');
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/kms/key-providers/ed25519-provider.ts]: Done

File: src/kms/key-providers/ed25519-provider.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    const signature = await ed25519.sign(data, privateKey);
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/kms/key-providers/ed25519-provider.ts]: Done

File: src/kms/key-providers/ed25519-provider.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    const publicKey = ed25519.getPublicKey(privateKey);
+    const isValid = await ed25519.verify(signature, data, publicKey);
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/kms/key-providers/ed25519-provider.ts]: Removed

File: tests/kms/mks.test.ts
Comment (elias-garcia) [inline tests/kms/mks.test.ts:1]: The name of the file looks wrong, I think that it should be `kms.test.ts`

Comment (amonsosanz) [reply tests/kms/mks.test.ts:1]: Yep, fixed!

**Review by elias-garcia — APPROVED** [review]

**Pull request merged**